roles/deploy/README.md

```diff
@@ -0,0 +1,55 @@
+# Deploy
+
+## Overview
+The `deploy` role will read the OSPF facts from the provided/default or remote inventory and deploy the changes onto the appliances. It enables users to read the config as `host_vars` from a local or remote data store and deploy network configurations if any changes are found. This role is designed to ensure that network configurations remain consistent and compliant across devices.
+
+## Features
+- Read network configuration data from local or remote repositories.
+- Deploy configurations only when changes are detected.
+- Support for integration with Git repositories for version control.
+
+## Variables
+
+| Variable Name        | Default Value | Required | Type | Description                                                   | Example |
+|:---------------------|:-------------:|:--------:|:----:|:-------------------------------------------------------------|:-------:|
+| `ansible_network_os` | `""`          | no      | str  | Network OS to be used during deploy.                    | `"cisco.nxos.nxos"` |
+| `resources`          | `[all]`          | no      | list | List of resources to be deployed (e.g., ospfv2, ospfv3).   | `['interfaces', 'vlans']` |
```

Comment:
I believe by default it will include ospf related resources for specific OS instead of `all`